kani-compiler/src/kani_middle/transform/mod.rs

```diff
@@ -118,6 +128,21 @@ impl BodyTransformation {
         }
     }
 
+    /// Run all global passes and store the results in a cache that can later be queried by `body`.
+    pub fn run_global_passes(
+        &mut self,
+        tcx: TyCtxt,
+        starting_items: &[MonoItem],
+        instances: Vec<Instance>,
+        call_graph: CallGraph,
+    ) {
+        let mut global_passes = std::mem::take(&mut self.global_passes);
```

Comment:
The other option would be to extract the cache and pass it here instead.

Reply:
I think this is less desirable because we would need to make sure the cache is already populated versus populating it on-demand

Reply:
actually, does global passes need to be part of the Transformer at all?

Reply:
I don't think so, actually -- I believe we run them only in one place so I could change that